apps/nextra/pages/en/build/create-aptos-dapp/templates/custom-indexer.mdx

```diff
@@ -0,0 +1,230 @@
+
+## Deploy indexer to a live server
+
+We recommend using Google Cloud Run to host the indexer, Secret Manager to store `config.yaml` and Artifact Registry to store the indexer docker image.
+
+### Build the docker image locally and run the container locally
+
+Build the docker image targeting linux/amd64 as we will eventually push the image to Artifact Registry and deploy it to Cloud Run, which only supports linux/amd64.
+
+```sh
+docker build --platform linux/amd64 -t indexer .
+```
+
+You can run the docker container locally to make sure it works. Mac supports linux/amd64 emulation so you can run the x86 docker image on Mac.
+
+```sh
+docker run -p 8080:8080 -it indexer
+```
+
+### Push the locally build docker image to Artifact Registry
+
+Login to google cloud
+
+```sh
+gcloud auth login
+```
+
+Create a repo in the container registry and push to it. You can learn more about publishing to Artifact Registry on their [docs](https://cloud.google.com/artifact-registry/docs/docker/pushing-and-pulling#pushing).
+
+Authorize docker to push to Artifact Registry. Set the default region below to your region.
+
+```sh
+# update us-west2 to your region, you can find it in google cloud
+gcloud auth configure-docker us-west2-docker.pkg.dev
+```
+
+Tag the docker image.
+
+```sh
+# update us-west2 to your region, you can find it in google cloud
+docker tag indexer us-west2-docker.pkg.dev/google-cloud-project-id/repo-name/indexer
+```
+
+Push to the Artifact Registry.
+
+```sh
+# update us-west2 to your region, you can find it in google cloud
+docker push us-west2-docker.pkg.dev/google-cloud-project-id/repo-name/indexer
+```
+
+### Upload the config.yaml file to Secret Manager
+
+Go to secret manager and create a new secret using the `config.yaml` file. Please watch this video walkthrough carefully: https://drive.google.com/file/d/1bbwe617fqM31swqc9W5ck8G8eyg3H4H2/view
+
+### Run the container on Cloud Run
+
+Please watch this video walkthrough carefully and follow the exact same setup: https://drive.google.com/file/d/1JayWuH2qgnqOgzVuZm9MwKT42hj4z0JN/view.
+
+Go to the Cloud Run dashboard, create a new service, then select the container image from Artifact Registry. Make sure to add a volume to load the config.yaml file from Secret Manager, then mount the volume to the container.
+
+You can learn more about Cloud Run in their [docs](https://cloud.google.com/run/docs/quickstarts/deploy-container).
+
+**NOTE**: Always allocate CPU so it always runs instead of only run when there is traffic. Min and max instances should be 1.
+
+## Re-indexing
+
+**WARNING**: Do not ever try to backfill the data, as logic like point calculation is incremental. If you backfill (e.g. processing same event twice), you will get wrong point data. So please always revert all migrations and re-index from the first tx your contract deployed.
```

Comment:
> as logic like point calculation is incremental

I dont understand this. Maybe we should use a callout for this too and put it much higher since this seems pretty important